spec/API_specification/array_api/fourier_transform_functions.py

```diff
@@ -0,0 +1,468 @@
+    """
+    Computes the n-dimensional inverse discrete Fourier transform.
+
+    Parameters
+    ----------
+    x: array
+        input array. Should have a floating-point data type.
+    s: Sequence[int]
+        size of each transformed axis of the output. If
+
+        - ``s[i]`` is greater than the size of the input array along a corresponding axis (dimension) ``j``, the corresponding axis of the input array must be zero-padded such that the axis has size ``s[i]``.
+        - ``s[i]`` is less than the length of the input array, each axis ``i`` of the input array must be trimmed to the length ``s[i]``.
+
+        If not provided, the length of the transformed axes of the output must equal the length of the input along the axes specified by ``axes``. Default: ``None``.
+    axes: Sequence[int]
+        axes over which to compute the inverse Fourier transform. If not specified, the last ``len(s)`` axes are used, or all axes if ``s`` is not specified either. Default: ``None``.
+    norm: Literal['backward', 'ortho', 'forward']
+        specify the normalization mode. Should be one of the following modes:
+
+        - ``'backward'``: normalize by ``1/n``.
+        - ``'ortho'``: normalize by ``1/sqrt(n)`` (i.e., make the FFT orthonormal).
+        - ``'forward'``: no normalization.
+
+        where ``n`` equals ``prod(s)``, the logical FFT size.
+
+        Default: ``'backward'``.
+
+    Returns
+    -------
+    out: array
+        an array transformed along the axes (dimension) indicated by ``axes``. The returned array must have a complex floating-point data type determine by :ref:`type-promotion`.
+    """
+
+
+def rfft(x: array, /, *, n: Optional[int] = None, axis: int = -1, norm: Literal['backward', 'ortho', 'forward'] = 'backward') -> array:
+    """
+    Computes the one-dimensional discrete Fourier transform for real-valued input.
+
+
+    .. note::
+       Applying the one-dimensional inverse discrete Fourier transform for real-valued input to the output of this function must return the original (i.e., non-transformed) input array within numerical accuracy (i.e., ``irfft(rfft(x), n=x.shape[axis]) == x``), provided that the transform and inverse transform are performed with the same normalization mode..
+
+    Parameters
+    ----------
+    x: array
+        input array. Should have a real-valued floating-point data type.
+    n: int
+        length of the transformed axis of the **input**. If
+
+        - ``n`` is greater than the length of the input array, the input array must be zero-padded such that the input array has length ``n``.
+        - ``n`` is less than the length of the input array, the input array must be trimmed to length ``n``.
+
+        If not provided, the length of the transformed axis of the output must equal the length of the input along the axis specified by ``axis``. Default: ``None``.
```

Comment:
As noted in my previous comments, this is incorrect. After determining `n` (trimmed, padded, or as-is) the output length will always be n/2+1 (with variation depending on `n` is even or odd).